x/execution/internal/keeper/keeper.go

```diff
@@ -86,10 +92,21 @@ func (k *Keeper) Create(ctx sdk.Context, msg types.MsgCreateExecution) (*executi
 	if err != nil {
 		return nil, err
 	}
+	priceBz, err := k.cdc.MarshalBinaryLengthPrefixed(msg.Price)
+	if err != nil {
+		return nil, err
+	}
+
 	if !ctx.IsCheckTx() {
 		M.InProgress.Add(1)
 	}
+
+	if err := k.supplyKeeper.DelegateCoinsFromAccountToModule(ctx, msg.Signer, types.ModuleName, msg.Price); err != nil {
+		return nil, err
+	}
+
 	store.Set(exec.Hash, value)
+	store.Set(append(exec.Hash, []byte("_price")...), priceBz)
```

Comment:
what is this?
do we need to keep the price here? can't we just add it to the execution struct (`exec.Price`)?

Reply:
we can. This it the second option I wanted to discuss.

Reply:
let's add the price to the execution itself. like this it will be available in the ABCI and thus easy for client to display it ;)